lib/js-file.js

```diff
@@ -543,7 +546,196 @@ JsFile.prototype = {
         });
 
         return lines;
+    },
+
+    /**
+     * Builds token list using both code tokens and comment-tokens.
+     *
+     * @returns {Object[]}
+     * @private
+     */
+    _buildTokenList: function(codeTokens, commentTokens) {
+        var result = [];
+        var codeQueue = codeTokens.concat();
+        var commentQueue = commentTokens.concat();
+        while (codeQueue.length > 0 || commentQueue.length > 0) {
+            if (codeQueue.length > 0) {
+                if (commentQueue.length > 0 && commentQueue[0].range[0] < codeQueue[0].range[0]) {
+                    result.push(commentQueue.shift());
+                } else {
+                    result.push(codeQueue.shift());
+                }
+            } else {
+                result.push(commentQueue.shift());
+            }
+        }
+        return result;
+    },
+
+    /**
+     * Adds `_isComment` flat to the comment-tokens.
+     *
+     * @param {Object[]} commentTokens
+     * @private
+     */
+    _markCommentTokens: function(commentTokens) {
```

Comment:
I see. OK, then. Perhaps do this step inside buildTokenList so that an additional loop can be saved?